packages/react/src/converter/react-wrapper.tsx

```diff
@@ -1,74 +1,251 @@
-        eventName = eventName.charAt(0).toLowerCase() + eventName.slice(1);
-        this.ref.current.addEventListener(eventName, e => this.props[prop](e));
-      }
+    /**
+     * Updates element properties correctly setting properties
+     * on every update. Note, this does not include mount.
+     */
+    componentDidUpdate(old: ComponentProps) {
+      this._updateElement(old);
     }
 
-    _updateCustomElementProperty(prop) {
-      if (this.ref.current) {
-        this.ref.current[prop] = this.props[prop];
+    /**
+     * Renders the custom element with a `ref` prop which allows this
+     * component to reference the custom element.
+     *
+     * Standard attributes are passed to React and element properties and events
+     * are updated in componentDidMount/componentDidUpdate.
+     *
+     */
+    render() {
+      // Since refs only get fulfilled once, pass a new one if the user's
+      // ref changed. This allows refs to be fulfilled as expected, going from
+      // having a value to null.
+      const userRef = this.props.__forwardedRef as React.Ref<unknown>;
+      if (this._ref === undefined || this._userRef !== userRef) {
+        this._ref = (value: I | null) => {
+          if (this._element === null) {
+            this._element = value;
 
-        // if prop value is a string we assume to set the attribute on custom element
-        if (typeof this.props[prop] === 'string') {
-          this.ref.current.setAttribute(prop, this.props[prop]);
+            Object.defineProperty(this._element, 'nativeElement', {
+              get: () => new Promise(r => r(this._element)),
+            });
+          }
+          if (userRef !== null) {
+            setRef(userRef, value);
+          }
+          this._userRef = userRef;
+        };
+      }
+      // Filters class properties out and passes the remaining
+      // attributes to React. This allows attributes to use framework rules
+      // for setting attributes and render correctly under SSR.
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const props: any = { ref: this._ref };
+      // Note, save element props while iterating to avoid the need to
+      // iterate again when setting properties.
+      this._elementProps = {};
+      for (const [k, v] of Object.entries(this.props)) {
+        if (elementClassProps.has(k)) {
+          this._elementProps[k] = v;
+        } else {
+          // React does *not* handle `className` for custom elements so
+          // coerce it to `class` so it's handled correctly.
+          props[k === 'className' ? 'class' : k] = v;
         }
       }
+      return createElement(tagName, props);
     }
-  };
-}
+  }
+
+  const element = React.forwardRef((props?: UserProps, ref?: React.Ref<unknown>) =>
+    createElement(ReactComponent, { ...props, __forwardedRef: ref } as any, props?.children)
+  );
+
+  element.displayName = `${componentName}`; // edit to preserve component name in DevTools - Cory
```

Comment:
I added a comment to any lines I specifically changed from the original lit wrapper.